core/src/stages/generate_pose.cpp

```diff
@@ -40,13 +40,47 @@
 #include <moveit/planning_scene/planning_scene.h>
 #include <rviz_marker_tools/marker_creation.h>
 
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <tf2/convert.h>
+
+#include <chrono>
+
+namespace {
+// TODO(henningkayser): support user-defined random number engines
+std::random_device rd_;
+std::mt19937 gen_(rd_());
+}
+
 namespace moveit {
 namespace task_constructor {
 namespace stages {
 
 GeneratePose::GeneratePose(const std::string& name) : MonitoringGenerator(name) {
 	auto& p = properties();
 	p.declare<geometry_msgs::PoseStamped>("pose", "target pose to pass on in spawned states");
+	p.declare<size_t>("max_solutions", 20,
+	                  "limit of the number of spawned solution in case randomized sampling is enabled");
+	p.property("timeout").setDefaultValue(10.0 /* seconds */);
```

Comment:
10 seconds to generate random poses seems ridiculous.
Limit this to max 1 second?